script/translations/develop.py

```diff
@@ -69,7 +69,7 @@ def substitute_translation_references(integration_strings, flattened_translation
 
 def substitute_reference(value, flattened_translations):
     """Substitute localization key references in a translation string."""
-    matches = re.findall(r"\[\%key:((?:[a-z0-9-_]+|[:]{2})*)\%\]", value)
+    matches = re.findall(r"\[\%key:((?:[a-z0-9-_]+|[:]{2})*+)\%\]", value)
```

Comment:
```suggestion
    matches = re.findall(r"\[\%key:((?:[a-z0-9-_]+|[:]{2})+)\%\]", value)
```

Reply:
With this change, we no longer match this: `[%%]`, we still match `[%::%]` though, so I don't think it's useful change.
Moreover, it doesn't fix the problem this PR is fixing: https://regex101.com/r/EAdx7P/1

Reply:
In that case, this might make more sense:

`\[\%key:[a-z0-9_]+(::([a-z0-9-_])+)*\%\]`

It also removes the backtracking, making it faster.

![CleanShot 2023-10-22 at 16 36 02@2x](https://github.com/home-assistant/core/assets/195327/0e31fd4e-4dbe-478f-bc4f-aaeb95f4e279)

(PS: I removed the `-` from the initial part of the key, as I don't believe we have those cases. For the second case I kept it in, as I think we still have a few cases that require these in our codebase).

Reply:
The point of this regex is not to validate the reference, we do that later and fail on any reference that does not exist. Hence, I'm not sure if it's good to make the regex more targeted, because we might not warn on references with typos, maybe the regex should in fact be simplified, not made more complicated? 🤔

For example, `[%key:component:bluetooth::test%]` won't be matched, but perhaps it *should* be matched by the regex so there's a warning later?

Reply:
> ut perhaps it should be matched by the regex so there's a warning later

Should? In which case?

Reply:
Hmm, OK, the validation is done by hassfest, not here. Alright, your changes is probably better then 👍